src/Aspire.Hosting.AWS/Provisioning/Provisioners/AWSProvisioner.cs

```diff
@@ -0,0 +1,167 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.AWS.Provisioning;
+
+internal sealed class AWSProvisioner(
+    DistributedApplicationExecutionContext executionContext,
+    IServiceProvider serviceProvider,
+    ResourceNotificationService notificationService,
+    ResourceLoggerService loggerService) : IDistributedApplicationLifecycleHook
+{
+    public async Task BeforeStartAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken = default)
+    {
+        if (executionContext.IsPublishMode)
+        {
+            return;
+        }
+
+        var awsResources = appModel.Resources.OfType<IAWSResource>().ToList();
+        if (awsResources.Count == 0)
+        {
+            return;
+        }
+
+        static IAWSResource? SelectParentAWSResource(IResource resource) => resource switch
```

Comment:
My style preference is to use private method instead of local methods especially when defining more then one. I feel like with this many local methods defined it makes it harder to quickly see what this function does. At the very least add the local methods to the end of the method body so it is easy to see what the function's logic is versus the utility methods the method relies on.

Reply:
This was included due to the adoption of the Azure Provisioner style. I'll change this as it is not my preferred style as while.